### PR TITLE
Deprecate incorrect import locations

### PR DESCRIPTION
**Pull request by oscargus**
## PR Summary

Related to #23554

`TextPath` has based on the comment been imported from `text` earlier. `AxislineStyle` does not have such a comment, but is related enough to guess that there may be code importing from `axislines` rather than `axisline_style`. So to be on the safe side.

(One should possibly add that to `axisartist/__init__.py` and encouraging importing from there...)

## PR Checklist

**Tests and Styling**
- [N/A] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [N/A] New features are documented, with examples if plot related.
- [N/A] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [x] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [N/A] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

File: doc/api/next_api_changes/deprecations/23565-OG.rst [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Alternative import location for ``TextPath`` and ``AxislineStyle``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+These two classes have historically sometimes been imported from incorrect
+locations. This is now deprecated and `.TextPath` should be imported from
```
Comment (timhoffm) [inline doc/api/next_api_changes/deprecations/23565-OG.rst:5]: I'm not sure we should do this. If we deprecate we break a lot of user code eventually.

https://github.com/search?q=%22from+matplotlib.textpath+import+TextPath%22&type=code
textpath -> 825 hits

https://github.com/search?q=%22from+matplotlib.text+import+TextPath%22&type=code
text -> 1783 hits

Could do a pending deprecation.

I also don't know which way I would deprecate. Sure, TextPath is in TextPath (alongside with TextToPath). But is it really necessary to have that many modules as API surface. I can see an argument that these are text-like enough to be accessed via the `text` module. - Which would long-term mean making `textpath` private, or moving the contents to text directly.

Comment (anntzer) [reply doc/api/next_api_changes/deprecations/23565-OG.rst:5]: Agreed that if we really want to pick one module (which is not clear), I'd rather make text the "official" one.

Comment (oscargus) [reply doc/api/next_api_changes/deprecations/23565-OG.rst:5]: OK! But then maybe we should at least make sure that `TextPath` is documented as being in `text` and not in `textpath`? https://matplotlib.org/devdocs/api/textpath_api.html

Comment (oscargus) [reply doc/api/next_api_changes/deprecations/23565-OG.rst:5]: `textpath` was made its own module in 2009. Although it doesn't really mean that it was fullt used for its purposes.

Comment (oscargus) [reply doc/api/next_api_changes/deprecations/23565-OG.rst:5]: See #23576 for the `textpath` module.

Comment (oscargus) [reply doc/api/next_api_changes/deprecations/23565-OG.rst:5]: As a preparation for the dev call:

`textpath` 25 hits (5 matplotlib, 8 Visualization book)
https://grep.app/search?q=from%20matplotlib.textpath%20import%20TextPath

`text` 13 hits (3 matplotlib, 3 matplotlib-cn, 4 Visualization book)
https://grep.app/search?q=from%20matplotlib.text%20import%20TextPath

**Comment by timhoffm** [comment]
I have to think about this. Let's target 3.7.

**Comment by tacaswell** [comment]
Closed in favor of #23576 which picks `matplotlib.text` as the canonical "correct" import location .

**Pull request closed without merging**